File: saydigits.cpp
#include <iostream>
using namespace std;

void say_digits(int n,string arr[]){
    //base case
    if (n==0){
        return;}
    int digit=n% 10;
    n=n/10;
    say_digits(n,arr);
    //processing
    cout <<arr[digit] <<" ";};

int main(){
    int n=1089;
    string arr[10]={"zero","one","two","three","four","five","six","seven","eight","nine"};
    say_digits(n,arr);
    return 0;}

File: saydigits.py
#say digits recursive solution where parameter n is string

def saydigits(n):
    nums=['zero','one','two','three','four','five','six','seven','eight','nine']
    if n=='':
        return ''
    return nums[int(n[0])]+' '+saydigits(n[1:])
  

def main():
    n=input('enter number: ')
    print(saydigits(n))
main()